lib/chef/provisioning/aws_driver/driver.rb

```diff
@@ -893,25 +896,133 @@ def default_ami_for_region(region, criteria = {})
+      if machine_spec.reference[:winrm_ssl_cert]
+        FileUtils.mkdir_p(Chef::Config.trusted_certs_dir)
+        filename = File.join(Chef::Config.trusted_certs_dir, "#{machine_spec.name}.crt")
+        if File.exists?(filename)
+          Chef::Log.warn("Existing cert for #{winrm_subject} in #{filename}")
+        else
+          Chef::Log.warn("Adding certificate for #{winrm_subject} in #{filename}")
+          File.open(filename, File::CREAT|File::TRUNC|File::RDWR, 0644) do |f|
+            f.print(machine_spec.reference[:winrm_ssl_cert])
+          end
+        end
+        winrm_options[:ca_trust_path] = filename
+      end
       Chef::Provisioning::Transport::WinRM.new("#{endpoint}", type, winrm_options, {})
+      binding.pry
+      # in order for ssl to work, we need to tell ssl that it's ok that ssl_subject doesn't
+      # match the ip
+      #[19] pry(#<Chef::Provisioning::AWSDriver::Driver>)> endpoint.split('/')[2].split(':').first
+      #  => "10.113.70.104"
+      #  [20] pry(#<Chef::Provisioning::AWSDriver::Driver>)> machine_spec.reference[:winrm_ssl_subject]
+      #    => "IP-0A714668"
+      #
+      # [1] pry(#<Chef::Provisioning::AWSDriver::Driver>)> Chef::Provisioning::Transport::WinRM.new("#{endpoint}", type, winrm_options, {}).execute('hostname')
+      # OpenSSL::SSL::SSLError: SSL_connect returned=1 errno=0 state=error: certificate verify failed
+      # from /home/hh/.rvm/gems/ruby-2.2.0/gems/httpclient-2.6.0.1/lib/httpclient/session.rb:307:in `connect'
+      # [2] pry(#<Chef::Provisioning::AWSDriver::Driver>)> wtf!
+      # Exception: OpenSSL::SSL::SSLError: SSL_connect returned=1 errno=0 state=error: certificate verify failed
+      # --
+      # 0: /home/hh/.rvm/gems/ruby-2.2.0/gems/httpclient-2.6.0.1/lib/httpclient/session.rb:307:in `connect'
+      # 1: /home/hh/.rvm/gems/ruby-2.2.0/gems/httpclient-2.6.0.1/lib/httpclient/session.rb:307:in `ssl_connect'
+      # 2: /home/hh/.rvm/gems/ruby-2.2.0/gems/httpclient-2.6.0.1/lib/httpclient/session.rb:755:in `block in connect'
+      # 3: /home/hh/.rvm/rubies/ruby-2.2.0/lib/ruby/2.2.0/timeout.rb:89:in `block in timeout'
+      # 4: /home/hh/.rvm/rubies/ruby-2.2.0/lib/ruby/2.2.0/timeout.rb:99:in `call'
+      # 5: /home/hh/.rvm/rubies/ruby-2.2.0/lib/ruby/2.2.0/timeout.rb:99:in `timeout'
+      # 6: /home/hh/.rvm/rubies/ruby-2.2.0/lib/ruby/2.2.0/timeout.rb:125:in `timeout'
+      # 7: /home/hh/.rvm/gems/ruby-2.2.0/gems/httpclient-2.6.0.1/lib/httpclient/session.rb:746:in `connect'
+      # 8: /home/hh/.rvm/gems/ruby-2.2.0/gems/httpclient-2.6.0.1/lib/httpclient/session.rb:612:in `query'
+      # 9: /home/hh/.rvm/gems/ruby-2.2.0/gems/httpclient-2.6.0.1/lib/httpclient/session.rb:164:in `query'
+      # [3] pry(#<Chef::Provisioning::AWSDriver::Driver>)> winrm_options
+      # => {:user=>"Administrator",
+      #  :pass=>"(xntd8f=-HNnuJ3",
+      #   :disable_sspi=>false,
+      #    :basic_auth_only=>false,
+      #     :no_ssl_peer_verification=>false,
```

Comment:
you want `:no_ssl_peer_verification` to be `true` set in `machine_options[:winrm_no_ssl_peer_verification]`